osm2gtfs/creators/stops_creator.py

```diff
@@ -18,85 +15,94 @@ def __repr__(self):
+        This function adds a single Stop or Station object as a stop to GTFS.
+        It can be overridden by custom cretors to change how stop_ids are made
+        up.
+
+        :return stop_id: A string with the stop_id in the GTFS
+        """
+        try:
+            parent_station = stop.get_parent_station()
+        except AttributeError as e:
+            parent_station = ""
+
+        # Send stop_id creation through overridable function
+        gtfs_stop_id = self._define_stop_id(stop)
+
+        # Save defined stop_id to the object for furhter use in other creators
+        stop.set_stop_id(gtfs_stop_id)
+
+        # Set stop name
+        stop_name = self._define_stop_name(stop)
+
+        # Collect all data together for the stop creation
+        field_dict = {'stop_id': self._define_stop_id(stop),
+                      'stop_name': stop_name,
+                      'stop_lat': float(stop.lat),
+                      'stop_lon': float(stop.lon),
+                      'location_type': stop.location_type,
+                      'parent_station': parent_station
+                      }
 
         # Add stop to GTFS object
-        stop = transitfeed.Stop(field_dict=stop_dict)
-        feed.AddStopObject(stop)
-        return stop
+        feed.AddStopObject(transitfeed.Stop(field_dict=field_dict))
 
-    def add_stops_to_routes(self, data):
-        routes = data.routes
-        stops = data.stops
+        if type(stop_name).__name__ == "unicode":
+            stop_name = stop_name.encode('utf-8')
 
-        # Loop through routes
-        for ref, route in routes.iteritems():
-            # Replace stop ids with Stop objects
-            self._fill_stops(stops, route)
+        print("Added stop: " + stop_name +
+              " - " + stop.osm_url)
 
-        data.routes = routes
-        return
+        # Return the stop_id of the stop added
+        return field_dict['stop_id']
 
-    def _fill_stops(self, stops, route):
+    def _define_stop_id(self, stop):
         """
-        Fill a route object with stop objects for of linked stop ids
+        This function returns the GTFS stop id to be used for a stop.
+        It can be overridden by custom cretors to change how stop_ids are made
+        up.
+
+        :return stop_id: A string with the stop_id for use in the GTFS
         """
-        if isinstance(route, Route):
-            i = 0
-            for stop in route.stops:
-                # Replace stop id with Stop objects
-                # TODO: Remove here and use references in TripsCreatorFenix
-                route.stops[i] = self._get_stop(stop, stops)
-                i += 1
-
-        elif isinstance(route, RouteMaster):
-            for route_variant_ref, route_variant in route.routes.iteritems():
-                self._fill_stops(stops, route_variant)
 
+        if "gtfs_stop_id" in stop.tags:
+            #  Use a GTFS stop_id coming from OpenStreetMap data
+            return stop.tags['gtfs_stop_id']
```

Comment:
Is the usage of `gtfs_stop_id` in OpenStreetMap documented?

Reply:
No, it is not. If people here are very strong on taking it out. I'm fine with it. I just think it is a nice feature, especially when GTFS already exist in an area and the stops should map to those ids.

Reply:
I don't see the use cases for defining their own stop IDs, but if someone needs it, I'm fine with it. At least I would document it somewhere in the osm2gtfs wiki.

Reply:
In Grande Vitória, most bus stops have a `ref` defined by municipal or operator, usually posted on the bus stop sign, all known such references are mapped with the `ref` tag. This could be used instead of automatically get the OSM object ID. Making an independent `gifts_stop_id` tag will only duplicate data and result in data not being added.

Reply:
Most people who work with GTFS appear to be  kind of holy about their ids, and I can understand, because it is the reference to the thing. Yes, @Skippern, `ref` seems to be the correct way to go, maybe even also supportin `ref:gtfs` for special needs. `gtfs_stop_id` was a bad idea. Thanks, for noticing, will change!

Reply:
Changed, accordingly.